### PR TITLE
main promo banner

### LINKED ISSUE #1178: Main promo banner
Move the styling of the main promo banner to Yari

- Cleanup SASS to avoid unnecessary duplication
- Implement base banner style
- Implement basic theme
- Document in Storybook

### PR DESCRIPTION
**Pull request by schalkneethling**
Update style for the main promo banner. Remove the old header and replace it with the latest version.

fix #1178

### CODE REVIEW

**Review by peterbe — CHANGES_REQUESTED** [review]
The banners all seem to work and they look crisp. 
But I'm skeptical towards the `stylelint` integration and all the other unrelated changes to the `sass` and `prettier` and `node-sass-chokidar`. If these are important and necessary things we should put them into master first. 
Also, now there's a `stylelintrc` file but it's never used for anything. In CI we trigger `lint-check.sh` which runs `yarn prettier-check` but that's it.

File: package.json [hunk 1]
```diff
@@ -112,12 +110,21 @@
     "jest-junit-reporter": "^1.1.0",
     "jest-puppeteer": "4.4.0",
     "jsdom": "^16.4.0",
+    "node-sass-chokidar": "^1.5.0",
     "nodemon": "2.0.4",
     "pegjs": "^0.10.0",
-    "prettier": "2.1.1",
+    "prettier": "^2.1.1",
```
[the remark this answers is not in the document]
Comment (schalkneethling) [reply package.json:116]: Not sure. The only thing I can think is that when I installed the bits for stylelint and I might have included prettier in the `yarn add -D` list and this time it changed it from a fixed version to `^`

File: client/src/banners/active-banner.tsx [hunk 2]
```diff
@@ -157,6 +152,23 @@ function Banner(props: BannerProps) {
 
 const DEVELOPER_NEEDS_ID = "developer_needs";
 const SUBSCRIPTION_ID = "subscription_banner";
+const USER_INTERVIEWS_ID = "user_interviews";
+
+function UserInterviewsBanner() {
```
[the remark this answers is not in the document]
Comment (schalkneethling) [reply client/src/banners/active-banner.tsx:157]: Ah yes, I used a couple of different variations for testing. I will cleanup and only keep subscriptions and the developer needs banners

File: package.json [hunk 3]
```diff
@@ -112,12 +110,21 @@
...
     "pretty-quick": "3.0.0",
     "puppeteer": "2",
     "react-is": "^16.13.1",
+    "sass": "^1.26.10",
```
[the remark this answers is not in the document]
Comment (schalkneethling) [reply package.json:120]: We should use just plain `sass`. I can make that change but, I reckon I should make it post this PR when I also move the stylelint bits to master?

**Comment by schalkneethling** [comment]
> But I'm skeptical towards the stylelint integration and all the other unrelated changes to the sass and prettier and node-sass-chokidar. If these are important and necessary things we should put them into master first.

Agreed. I am happy to do that after merging this one. Let me know if that works for you.

> Also, now there's a stylelintrc file but it's never used for anything. In CI we trigger lint-check.sh which runs yarn prettier-check but that's it.

Indeed but, that is because if I turn it on in a `pre-commit` hook or during `CI`, it will fail as a lot of the SASS does not yet follow all of the rules. It does poke me inside VSCode though and so, I am gixing things as I move along. The intent in the end is to add this to our existing CI linting in the end.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (schalkneethling) [reply package.json:118]: I have removed `node-sass` and `node-sass-chokidar` and confirmed that both `yarn dev` and Storybook still works as expected. With that, we are only using the single `sass` dependency and, can use SASS modules

[the remark this answers is not in the document]
Comment (peterbe) [reply package.json:118]: That's awesome! Did you check that that `yarn prepare-build` also worked?

[the remark this answers is not in the document]
Comment (schalkneethling) [reply package.json:118]: Yup, `yarn prepare-build` works as expected

**Review by peterbe — APPROVED** [review]
One thing that messes with my brain is that you've merged in `master` into `schalkneethling:1178-main-promo-banner` and you're now including that in your PR to `mdn:mdn-fiori`. Right?
Ideally, we should constantly keep the `mdn:mdn-fiori` update-to-date with `master` and when you make a branch *based* on `mdn-fiori` *don't* merge in `master` into it. 
It makes the PR diffs very hard to read but I think I know roughly what's going on. 

The most important thing is that the banners now look spiffy, and you've decided that `sass` is better than `node-sass` and you're gearing up to add `stylelint` to the CI flow.

**Pull request merged**